src/pip/_internal/req/req_install.py

```diff
@@ -898,32 +900,34 @@ def archive(self, build_dir):
                 shutil.move(archive_path, dest_file)
             elif response == 'a':
                 sys.exit(-1)
-        if create_archive:
-            zip = zipfile.ZipFile(
-                archive_path, 'w', zipfile.ZIP_DEFLATED,
-                allowZip64=True
-            )
+
+        if not create_archive:
+            return
+
+        with zipfile.ZipFile(
```

Comment:
I think creating the `ZipFile` object before the `with` statement, would be cleaner.

```py
zip = zipfile.ZipFile(
    archive_path, 'w', zipfile.ZIP_DEFLATED, allowZip64=True
)
with zip:
    ...
```

It probably makes sense to also change the name of the variable but that can be a follow up (or a good first issue). Let's file an issue after this merges.

Reply:
Done [here](https://github.com/pypa/pip/pull/6988/files#diff-47a4ece629e4698a5eaefea075d3cbdfR907-R909). I changed the name as well, just to not distract from our existing good first issues.